rclcpp_generic/CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+cmake_minimum_required(VERSION 3.5)
+project(rclcpp_generic)
+
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
+
+# Windows supplies macros for min and max by default. We should only use min and max from stl
+if(WIN32)
+  add_definitions(-DNOMINMAX)
+endif()
+
+find_package(ament_cmake REQUIRED)
```

Comment:
`rosbag2` also had `ament_cmake_ros` – it seems to build fine without it though. Do you know what the differences to `ament_cmake` are?

Reply:
ament_cmake_ros only provides a few handy cmake macros (from what I can tell - unfortunately there is no README lol). These macros are specifically `ament_add_ros_isolated_gmock`, `ament_add_ros_isolated_gtest`, and `ament_add_ros_isolated_pytest`, which I don't think we use here.